store/copr/batch_coprocessor.go

```diff
@@ -765,13 +809,31 @@ func (b *batchCopIterator) handleTask(ctx context.Context, bo *Backoffer, task *
 
 // Merge all ranges and request again.
 func (b *batchCopIterator) retryBatchCopTask(ctx context.Context, bo *backoff.Backoffer, batchTask *batchCopTask) ([]*batchCopTask, error) {
-	var ranges []kv.KeyRange
-	for _, ri := range batchTask.regionInfos {
-		ri.Ranges.Do(func(ran *kv.KeyRange) {
-			ranges = append(ranges, *ran)
-		})
+	if batchTask.regionInfos != nil {
+		var ranges []kv.KeyRange
+		for _, ri := range batchTask.regionInfos {
+			ri.Ranges.Do(func(ran *kv.KeyRange) {
+				ranges = append(ranges, *ran)
+			})
+		}
+		ret, err := buildBatchCopTasks(bo, b.store, []*KeyRanges{NewKeyRanges(ranges)}, b.req.StoreType, nil, 0, false, 0, nil)
+		return ret, err
+	}
+	// Retry Partition Table Scan
+	keyRanges := make([]*KeyRanges, 0, len(batchTask.PartitionTableRegions))
+	pid := make([]int64, 0, len(batchTask.PartitionTableRegions))
+	for _, trs := range batchTask.PartitionTableRegions {
+		pid = append(pid, trs.PhysicalTableId)
+		ranges := make([]kv.KeyRange, 0, len(trs.Regions))
+		for _, ri := range batchTask.regionInfos {
```

Comment:
it should be `for _, ri := range trs.Regions`?

Reply:
addressed.